### ISSUE TITLE
dsc_resource Crash on Error (powershell.rb)

### ISSUE THREAD

**Issue by jeremyciak** [chef/chef#11169]
## Description

After updating our Chef clients to version 16.10.17 we are now getting crashing errors on runs that would error out previously (e.g. - Chef Client version 16.4.41 and 16.2.73) but still allow the runs to complete and report back to the Chef Automate Server. It appears that this is occurring with errors experienced for dsc_resource resources.

## Chef Version

16.10.17

## Platform Version

Windows Server 2019

## Replication Case

```ruby
run_as_user = 'user.name'
run_as_password = 'P@$$w0rd!'
run_as_domain = 'DOMAIN'

ps_dsc_run_as_credential = ps_credential((run_as_domain + '\\' + run_as_user), run_as_password)

file_path = 'C:\blah\calc.exe'

dsc_resource "RemoteApp_Calculator" do
  resource :xRDRemoteApp
  module_name 'xRemoteDesktopSessionHost'
  module_version '2.0.0'
  property :CollectionName, 'RemoteApps'
  property :DisplayName, 'RemoteApp_Calculator'
  property :FilePath, file_path
  property :Alias, 'RemoteApp_Calculator'
  property :Ensure, 'Present'
  property :PsDscRunAsCredential, ps_dsc_run_as_credential
  action :run
end
```

If the 'FilePath' property of the `dsc_resource` is not a valid path, this causes the DSC execution to error. In previous Chef Client versions (e.g. - 16.4.41 and 16.2.73) this would manifest in the Chef Client log/output as an error in the run and the run would still complete. With version 16.10.17 these errors are not handled well and cause the run to crash completely.

I updated the resource block to have a file path validation guard and it works after doing that because it avoids the error scenario entirely:

```ruby
run_as_user = 'user.name'
run_as_password = 'P@$$w0rd!'
run_as_domain = 'DOMAIN'

ps_dsc_run_as_credential = ps_credential((run_as_domain + '\\' + run_as_user), run_as_password)

file_path = 'C:\blah\calc.exe'

dsc_resource "RemoteApp_Calculator" do
  resource :xRDRemoteApp
  module_name 'xRemoteDesktopSessionHost'
  module_version '2.0.0'
  property :CollectionName, 'RemoteApps'
  property :DisplayName, 'RemoteApp_Calculator'
  property :FilePath, file_path
  property :Alias, 'RemoteApp_Calculator'
  property :Ensure, 'Present'
  property :PsDscRunAsCredential, ps_dsc_run_as_credential
  action :run
  only_if do
    powershell_out(
      "Test-Path -Path '#{file_path}' -ErrorAction SilentlyContinue",
      user: run_as_user,
      password: run_as_password,
      domain: run_as_domain
    ).stdout.chomp().to_s.casecmp?('true')
  end
end
```

## Client Output

https://gist.github.com/jeremyciak/a2c1b025dbd1500bb2ece96f4e2d2e8f

**Comment by jeremyciak** [comment]
This issue is continuing to plague my Chef runs which yield no useful information in the logs when the failures occur. Is this slotted for an upcoming release I can look forward to? Has it been resolved in a release since this version (16.10.17)? I see that version 16.11.7 is available now... should I switch to that or would that not help with this?

**Comment by wheatevo** [comment]
Seeing similar segmentation fault issues in Chef 16.8.14.
